tests/e2e_tests.py

```diff
@@ -0,0 +1,364 @@
+        if os.path.exists(TestBase.STUDENTS_DIR_PATH):
+            for file in os.listdir(TestBase.STUDENTS_DIR_PATH):
+                os.remove(os.path.join(TestBase.STUDENTS_DIR_PATH, file))
+
+    def test_output_parse(self):
+        input_model = ModelGenerator.generate_input_model()
+        parsed_output = TestBase.parse_output(input_model)
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+
+        output_model = Output(**TestBase.read_from_file(f"{TestBase.STUDENTS_DIR_PATH}/{TestBase.get_docs(1)[0]}"))
+
+        self.assertEqual(output_model.studentDetails, parsed_output.studentDetails.__dict__)
+        for subject in range(len(parsed_output.subjectGrades)):
+            self.assertEqual(output_model.subjectGrades[subject], parsed_output.subjectGrades[subject].__dict__)
+        self.assertEqual(output_model.totalAvg, parsed_output.totalAvg)
+        self.assertEqual(output_model.birthDate, parsed_output.birthDate)
+        self.assertEqual(output_model.age, parsed_output.age)
+        self.assertEqual(output_model.gender, parsed_output.gender)
+        self.assertEqual(output_model.isGoodBehaviour, parsed_output.isGoodBehaviour)
+        self.assertEqual(output_model.age, input_model.age)
+        self.assertEqual(output_model.studentDetails['firstName'], input_model.studentDetails['firstName'])
+        self.assertEqual(output_model.studentDetails['lastName'], input_model.studentDetails['lastName'])
+        self.assertEqual(output_model.studentDetails['id'], input_model.studentDetails['id'])
+
+    def test_valid_file_name_capitalized(self):
+        input_model = ModelGenerator.generate_input_model()
+
+        input_model.studentDetails['firstName'] = 'test'
+        input_model.studentDetails['lastName'] = 'tester'
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+
+        self.assertEqual(['Test_Tester.txt'], TestBase.get_docs(1))
+
+    def test_valid_file_name_hyphen_lastname(self):
+        input_model = ModelGenerator.generate_input_model()
+
+        input_model.studentDetails['firstName'] = 'Test'
+        input_model.studentDetails['lastName'] = 'Ben Test'
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+
+        self.assertEqual(['Test_Ben-Test.txt'], TestBase.get_docs(1))
+
+    def test_send_same_body_twice(self):
+        input_model = ModelGenerator.generate_input_model()
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+        TestBase.send_body(self.rabbitmq_publisher, input_model)
+
+        self.assertTrue(1, len(TestBase.get_docs(1)))
+
+    def test_two_text_files_saved_success(self):
+        input_model_1 = ModelGenerator.generate_input_model()
+        input_model_2 = ModelGenerator.generate_input_model()
+
+        TestBase.send_body(self.rabbitmq_publisher, input_model_1)
+        TestBase.send_body(self.rabbitmq_publisher, input_model_2)
+
+    def test_send_non_json_body(self):
```

Comment:
test_send_non_json_body_failure or test_sent_invalid_format_failure
you forgot the failure